legion/pummaMM/taco-generated-tblis.cpp

```diff
@@ -0,0 +1,773 @@
+  launcher.add_region_requirement(RegionRequirement(c_vals_partition, 0, READ_ONLY, EXCLUSIVE, c_vals_parent, Mapping::DefaultMapper::VIRTUAL_MAP).add_field(c_vals_field_id));
+  launcher.tag = launcher.tag | TACOMapper::UNTRACK_VALID_REGIONS;
+  auto fm = runtime->execute_index_space(ctx, launcher);
+  fm.wait_all_results();
+
+
+  computePartitions.aPartition.indicesPartitions = std::vector<std::vector<Legion::LogicalPartition>>(2);
+  computePartitions.aPartition.denseLevelRunPartitions = std::vector<IndexPartition>(2);
+  computePartitions.aPartition.valsPartition = a_vals_partition;
+  computePartitions.aPartition.denseLevelRunPartitions[0] = a_dense_run_0_Partition_0;
+  computePartitions.bPartition.indicesPartitions = std::vector<std::vector<Legion::LogicalPartition>>(2);
+  computePartitions.bPartition.denseLevelRunPartitions = std::vector<IndexPartition>(2);
+  computePartitions.bPartition.valsPartition = b_vals_partition;
+  computePartitions.bPartition.denseLevelRunPartitions[0] = b_dense_run_0_Partition_0;
+  computePartitions.cPartition.indicesPartitions = std::vector<std::vector<Legion::LogicalPartition>>(2);
+  computePartitions.cPartition.denseLevelRunPartitions = std::vector<IndexPartition>(2);
+  computePartitions.cPartition.valsPartition = c_vals_partition;
+  computePartitions.cPartition.denseLevelRunPartitions[0] = c_dense_run_0_Partition_0;
+
+  return computePartitions;
+}
+
+void task_7(const Task* task, const std::vector<PhysicalRegion>& regions, Context ctx, Runtime* runtime) {
+  PhysicalRegion a_vals = regions[0];
+  LogicalRegion a_vals_parent = regions[0].get_logical_region();
+  PhysicalRegion b_vals = regions[1];
+  LogicalRegion b_vals_parent = regions[1].get_logical_region();
+  PhysicalRegion c_vals = regions[2];
+  LogicalRegion c_vals_parent = regions[2].get_logical_region();
+
+  int64_t iln = task->index_point[0];
+  task_7Args* args = (task_7Args*)(task->args);
+  Legion::FieldID a_vals_field_id = args->a_vals_field_id;
+  Legion::FieldID b_vals_field_id = args->b_vals_field_id;
+  Legion::FieldID c_vals_field_id = args->c_vals_field_id;
+
+  auto b_vals_ro_accessor = createAccessor<AccessorROdouble2>(b_vals, b_vals_field_id);
+  auto c_vals_ro_accessor = createAccessor<AccessorROdouble2>(c_vals, c_vals_field_id);
+  auto a_vals_rw_accessor = createAccessor<AccessorRWdouble2>(a_vals, a_vals_field_id);
+
+  auto aDomain = runtime->get_index_space_domain(ctx, a_vals.get_logical_region().get_index_space());
+  auto bDomain = runtime->get_index_space_domain(ctx, b_vals.get_logical_region().get_index_space());
+  auto cDomain = runtime->get_index_space_domain(ctx, c_vals.get_logical_region().get_index_space());
+  if (bDomain.get_volume() == 0 || cDomain.get_volume() == 0)
+    return ;
+
+  auto aShape = std::vector<tblis::len_type>(2);
+  aShape[0] = 1 + (aDomain.hi()[0] - aDomain.lo()[0]);
+  aShape[1] = 1 + (aDomain.hi()[1] - aDomain.lo()[1]);
```

Comment:
Ideally the shapes/strides arrays should be statically allocated, as in one of the following:

```c++
  tblis::len_type[] aShape = {
    1 + (aDomain.hi()[0] - aDomain.lo()[0]),
    1 + (aDomain.hi()[1] - aDomain.lo()[1]),
  };
```
or
```c++
  auto aShape = std::array<tblis::len_type, 2>();
  aShape[0] = 1 + (aDomain.hi()[0] - aDomain.lo()[0]);
  aShape[1] = 1 + (aDomain.hi()[1] - aDomain.lo()[1]);
```
Unfortunately, the IR seems too limited for these possibilities.

The first alternative doesn't work since there isn't a `{ … }` IR expression type. The second alternative technically works, but would require something like `"std::array<tblis::len_type, " + getOrder() + ">"`, which is a bit of a hassle and hence not done in this PR.

Reply:
Yeah, that's to leave out in this PR. Just make sure to add in a TODO that marks what needs to be done. However, I doubt that these dynamic allocations will affect performance at all, as the TBLIS computation should dominate.